### DISCUSSION TITLE
databend-query cluster run queries only on one node?

### DISCUSSION THREAD

**Discussion by Lif0820** [databendlabs/databend#12583] [Q&A]
I have a databend-query cluster with 3 nodes, which can be found in table `system.clusters`:
name           host                port
HnMW...     10.66.110.37    9091
opIBW...     10.66.110.39    9091
qJHdo...     10.66.110.38    9091

I send a slow query sql (named sql_A) to any one node (e.g. 10.66.110.38:3307) above by mysql client, 
and see cpu usage on linux shell by `top` command, only 10.66.110.37 cpu become busy, and the sql result cost about 10 seconds;

next, i send the sql_A query twice at same time, it stiil seems to only use 10.66.110.37 cpu, and the sql result cost almost 20 seconds;

so, my question is:
1. the databend-query cluster has 3 nodes, why it seems like only the first node in table `system.clusters` will be used? 
2. for one sql query which is sended to databend-query cluster, it will use only any one query-node? or may use all nodes to complete the query like Greenplum?

**Comment by wubx** [comment]
databend-query process query is cluster mode. 
There are some algorithms:
1. if query  user segment > cluster_size , every databend-query process segment/cluster_size 
    table segment can use:  select * from fuse_snapshot('db','table_name') limit 1;

2. if segment< cluster_szie , every databend-query processs blocks/cluster_size
    select * from fuse_snapshot('db','table_name') limit 1;

if the databend-query onlye one node process we need to make sure is there one segment and include one block.
